tests/hello_world_gles_compute.cpp

```diff
@@ -0,0 +1,823 @@
+/*
+ * Copyright 2019 The Emscripten Authors.  All rights reserved.
+ * Emscripten is available under two separate licenses, the MIT license and the
+ * University of Illinois/NCSA Open Source License.  Both these licenses can be
+ * found in the LICENSE file.
+ *
+ * Ray tracing with compute shader, inspired by
+ *
+ *   smallpt by Kevin Beason (http://kevinbeason.com/smallpt/)
+ *
+ * and
+ *
+ *   SmallptGPU by David Bucciarelli (http://davibu.interfree.it/opencl/smallptgpu/smallptGPU.html)
+ */
+
+#define GLFW_INCLUDE_ES31
+#include <GLFW/glfw3.h>
+#include <cmath>
+#include <cstdlib>
+#include <iostream>
+#include <limits>
+#include <random>
+#include <string>
+#include <vector>
+
+#if __EMSCRIPTEN__
+#include <deque>
+#include <emscripten.h>
+#include <functional>
+#endif
+
+// https://gcc.gnu.org/onlinedocs/cpp/Stringizing.html
+#define STR(s) #s
+#define XSTR(s) STR(s)
+
+#ifndef WIDTH
+#define WIDTH 320
+#endif
+#ifndef HEIGHT
+#define HEIGHT 240
+#endif
+
+#ifndef LOCAL_SIZE_X
+#define LOCAL_SIZE_X 1
+#endif
+#ifndef LOCAL_SIZE_Y
+#define LOCAL_SIZE_Y 1
+#endif
+
+#ifndef MAX_REGION_SIZE
+#define MAX_REGION_SIZE (WIDTH * HEIGHT * 4)
+#endif
+
+GLFWwindow *window;
+bool quit = false;
+
+GLuint compute_shader;
+GLuint vertex_shader;
+GLuint fragment_shader;
+GLuint compute_program;
+GLuint draw_program;
+GLuint rand_buffer;
+GLuint scene_buffer;
+GLuint texture;
+GLuint sampler;
+GLuint vertex_array;
+GLuint vertex_buffer;
+
+#if __EMSCRIPTEN__
+std::deque<std::function<void()>> queue;
+#endif
+
+void setup()
+{
+  if (!glfwInit()) {
+    exit(EXIT_FAILURE);
+  }
+  glfwWindowHint(GLFW_CLIENT_API, GLFW_OPENGL_ES_API);
+  glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 3);
+  glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 1);
+  window = glfwCreateWindow(WIDTH, HEIGHT, "SmallPT", nullptr, nullptr);
+  if (!window) {
+    exit(EXIT_FAILURE);
+  }
+  glfwMakeContextCurrent(window);
+
+  glClearColor(0.f, 0.f, 0.f, 1.f);
+
+  {
+    const GLchar *str =
```

Comment:
Is C++11 supported in these tests? Can you use `R"( ... )"`? (Same for other multiline strings in this file)

Reply:
I think c++11 is the default in clang now, so it should just work (if not, may need to add `-std=c++11` to the emcc flags, but in that case, maybe not worth it).

Reply:
I prefer the traditional style personally, though there isn't any solid reason. It's OK for me if someone else wants to submit a PR to modernize this.